pkg/stores/sqlpartition/listprocessor/processor.go

```diff
@@ -53,6 +56,61 @@ type Cache interface {
 	ListByOptions(ctx context.Context, lo informer.ListOptions, partitions []partition.Partition, namespace string) (*unstructured.UnstructuredList, int, string, error)
 }
 
+func k8sOpToRancherOp(k8sOp selection.Operator) (informer.Op, error) {
+	h := map[selection.Operator]informer.Op{
+		selection.Equals:       informer.Eq,
+		selection.DoubleEquals: informer.Eq,
+		selection.NotEquals:    informer.NotEq,
+		selection.In:           informer.In,
+		selection.NotIn:        informer.NotIn,
+		selection.Exists:       informer.Exists,
+		selection.DoesNotExist: informer.NotExists,
+		selection.LessThan:     informer.Lt,
+		selection.GreaterThan:  informer.Gt,
+	}
+	v, ok := h[k8sOp]
+	if ok {
+		return v, nil
+	}
+	return "", fmt.Errorf("unknown k8sOp: %s", k8sOp)
+}
+
+// Determine if the value field is surrounded by a pair of single- or double-quotes
+// This is a difference we implement from the CLI: if the target value of a (not) equal
+// test is quoted, we use the full string. Otherwise we do a substring match
+// (which is implemented as 'SELECT ... some-field ... LIKE "%VALUE%" ...' in the query)
+func isQuotedStringTarget(values []string) bool {
+	if len(values) != 1 || len(values[0]) == 0 {
+		return false
+	}
+	s1 := values[0][0:1]
+	if !strings.Contains(`"'`, s1) {
```

Comment:
When filtering for exact matches, do we want the string to be single quoted, double quoted, or support both? I am asking because the current behavior seem to be single quote:

https://github.com/rancher/steve/blame/d05a48f9cfa6a43a5fbb840cc5da00a92b229009/pkg/stores/sqlpartition/listprocessor/processor.go#L80

and IIUC this change would make it either single or double quote. Alas, we never documented the choice on the README.


@richard-cox do you have a preference?

@ericpromislow can you amend the README so that whatever we decide is documented please?

Reply:
Currently
- `'` denotes exact match
- no quotes denotes partial match

From my interpretation of docs the proposal would be the following?
- `'` denotes exact match
- `"` denotes partial match
- no quotes syntax has been removed?

I think ideally we would continue as currently. all other params, include those that contain text, do not require `"`

Reply:
No, quotes are still optional. And I've implemented single quotes indicate an exact match, and double quotes don't (they're just used to delimit the contents, but because of the syntax of k8s labels, they're never required because characters like spaces aren't allowed).

Reply:
Do we need double quotes at all?

Reply:
My first thought was it should probably be the operator that decides if its an exact match equality or a substring match. Something like: `=` is exact (stay closer to k8s), `=~` is substring. `!~` is not equal substring.

Imo we should have only one way per matching strategy, I don't see a reason to have both double quotes and no quotes denote substring match. That should simplify the logic a bit too.

Reply:
ok, double quotes are gone and trigger a syntax error